src/Chapter3.hs

```diff
@@ -479,6 +501,32 @@ After defining the city, implement the following functions:
    complicated task, walls can be built only if the city has a castle
    and at least 10 living people inside
 -}
+data City = City
+    { cityCastle            :: Castle
+    , cityCommunityBuilding :: CommunityBuilding
+    , cityHouses            :: [Int]
+    }
+
+data Castle
+    = None
+    | CastleWithWalls
+    | CastleWithoutWalls
+
+data CommunityBuilding
+    = Church
+    | Library
+
+data PeopleInHouse = One | Two | Three | Four
+
+buildCastle city = case cityCastle city of
+    None -> city { cityCastle = CastleWithoutWalls }
+    _ -> city
+
+buildHouse house city = city {cityHouses = house : cityHouses city}
+
+buildWalls city = case cityCastle city of 
+    CastleWithoutWalls -> if sum (cityHouses city) >= 10 then city { cityCastle = CastleWithWalls} else city
```

Comment:
One moment, here you count the number of houses, but in the task, we say about the number of people 🧑🏼‍🤝‍🧑🏼
So, you can write an additional function `PeopleInHouse -> Int` that pattern match on all constructors and return the number. After that, you would be able to count the number of people in all houses (some functions from the List Chapter would be useful here).